src/main/groovy/org/pih/warehouse/inventory/CycleCountTransactionCreator.groovy

```diff
@@ -0,0 +1,128 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.ValidationException
+import org.springframework.stereotype.Component
+
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.product.Product
+
+/**
+ * Component responsible for creating and persisting the Transactions that result from a cycle count.
+ */
+@Component
+class CycleCountTransactionCreator {
```

Comment:
what's the reason it is not just a regular service? I can see it calling GORM operations, so imho it should be a regular service and should be marked as `@Transactional`

Reply:
This is what convinced me: https://tedvinke.wordpress.com/2017/04/04/grails-anti-pattern-everything-is-a-service/

The flow I was imagining is controller -> service -> helper classes like this one, and since the service is already Transactional, not adding Transactional here would be an indicator that these classes shouldn't necessarily be called directly, it should always be through a service.

Reply:
sure we can discuss. And we can add Transactional here if we want. No need to make it a hard requirement that non-services always be called from services, I just felt it made the intended flow more explicit